### PR TITLE
Adding support for s390x in calculation of cpu_frequency

### PR DESCRIPTION
**Pull request by namrata-ibm**
PR for changing the GetCycleCounterFrequency for s390x to fix test `//tensorflow/core:platform_profile_utils_cpu_utils_test`.

Facing issue similar to power reported [here](https://github.com/tensorflow/tensorflow/issues/10450#issuecomment-307097621) .

On s390x too `cpu_frequency = -1` and hence below test fails:
```
[ RUN      ] CpuUtilsTest.CheckCycleCounterFrequency
2017-07-17 04:43:02.050579: F tensorflow/core/platform/profile_utils/cpu_utils_test.cc:57] Check failed: cpu_frequency > 0 (-1 vs. 0)
```

Extending the fix added for PPC raised via [PR ](https://github.com/tensorflow/tensorflow/pull/10522#issuecomment-311064869)

### CODE REVIEW

File: tensorflow/core/platform/profile_utils/cpu_utils.cc
Comment (gunan) [inline tensorflow/core/platform/profile_utils/cpu_utils.cc]: The block here looks exactly same as the block above. Shouldn't we just add `|| defined(__s390x__)` to the end of the `if` statement above?

File: tensorflow/core/platform/profile_utils/cpu_utils.h
Comment (gunan) [inline tensorflow/core/platform/profile_utils/cpu_utils.h]: same as above. Why not merge this with the `if` block?

File: tensorflow/core/platform/profile_utils/cpu_utils_test.cc
Comment (gunan) [inline tensorflow/core/platform/profile_utils/cpu_utils_test.cc]: same as above.

**Review by gunan — CHANGES_REQUESTED** [review]

**Comment by namrata-ibm** [comment]
@gunan, I saw a check for LE in the ppc commit and since s390x is big endian, thought of keeping the condition separate for s390x for avoiding any future conflicts. 
Would it be better to merge both in same if condition?

**Comment by gunan** [comment]
I would still merge these for the sake of keeping code cleaner.
Then we can worry about the added complexity (if needed) when adding ppc LE support.

**Comment by namrata-ibm** [comment]
@gunan, thank you for  your feedback.
Closing this PR; opened #12201

**Pull request closed without merging**
